scipy/interpolate/polyint.py

```diff
@@ -344,7 +343,7 @@ def _evaluate_derivatives(self, x, der=None):
             for i in range(1, n-k+1):
                 pi[i] = w[k+i-1]*pi[i-1] + pi[i]
                 cn[k] = cn[k] + pi[i, :, np.newaxis]*cn[k+i]
-            cn[k] *= factorial(k)
+            cn[k] *= float_factorial(k)
```

Comment:
IIRC the `scipy.special` factorial returns integer type and ***sometimes*** returns double type depending on the code path. So this could presumably change the output type for some codes, not sure if we should consider that as a backwards compat issue or a bug. 

Here is a link to the `special.factorial` code: 
https://github.com/scipy/scipy/blob/7dd0cd0365b2eda4484efed798000309e7649c77/scipy/special/_basic.py#L2365
this line is executed by default for `scipy.special.factorial` because `exact=False` is the default. The ufunc will return double.

Reply:
I think you've explained away your concern? This always returned float because it didn't pass `exact=True`.

Reply:
@peterbell10 hmm, seems like I did explain away that concern :) 

Also, I had a concern about the `exact=False` criterion returning different values but it looks like false branch uses the gamma function as an analytical continuation so results should match there too. 

I don't think `Nan`s will be propogated  in `math.factorial` and also `math.factorial` throws an error for me if a negative value is entered whereas in the `scipy.special` IIRC the returned values is `0` but in this case `k` comes from `range(1,k)` so also shouldn't be an issue. 

This changeset LGTM, marking approved.